dstep/translator/CodeBlock.d

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Mar 11, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module dstep.translator.CodeBlock;
+
+import std.array;
+import std.string;
+
+enum EndlHint
+{
+    /// Does nothing, useful for cases when the translator doesn't produce any ouput.
+    Empty,
+
+    /// Special node for grouping, puts an extra new line after grouped items, should not have spelling.
+    Group,
+
+    /// Single line item, should not have children.
+    SingleLine,
+
+    /// Multiline item, the children are indented.
+    MultiLine,
+
+    /// Puts braces if children takes more than single line, useful e.g for loops.
+    SubscopeWeak,
+
+    /// Always puts braces, useful e.g. for classes.
+    SubscopeStrong,
```

Comment:
Just remembered, the style convention in D is to start enum members with a lower case letter.